src/prefect/_internal/concurrency/threads.py

```diff
@@ -38,15 +39,17 @@ def start(self):
         """
         Start the worker thread.
         """
-        self.thread.start()
+        with self._lock:
+            if not self._started:
```

Comment:
If these are going to get re-used might be worthwhile to do "check, lock, check", to avoid the overhead of acquiring the lock.
```suggestion
    if not self._started:
        with self._lock:
            if not self._started:
```

Reply:
I don't think it should be called often. We do a guard like this in `submit` which is the most likely place for it to be called repeatedly https://github.com/PrefectHQ/prefect/pull/8830/files#diff-18ed573092916f8235b2a00fa32dc36b8aa5ca168e695d1e80038c561e59ae8dL50-R54